File: structural/python/composite/composite.py
from component import Component
from typing import List


class Composite(Component):

    def __init__(self) -> None:
        self._children: List[Component] = []

    def add(self, component: Component) -> None:
        self._children.append(component)
        component.parent = self

    def remove(self, component: Component) -> None:
        self._children.remove(component)
        component.parent = None

    def is_composite(self) -> bool:
        return True

    def operation(self) -> str:
        results = []
        for child in self._children:
            results.append(child.operation())
        return f"Branch({'+'.join(results)})"

File: structural/java/composite/src/Composite.java
import java.util.ArrayList;

public class Composite extends Component {

    private ArrayList<Component> children;

    public Composite(){
        this.children = new ArrayList<>();
    }

    @Override
    public void add(Component component) {
        this.children.add(component);
        component.setParent(this);
    }

    @Override
    public void remove(Component component) {
        this.children.remove(component);
        component.setParent(null);
    }

    @Override
    public String operation() {
        ArrayList<String> result = new ArrayList<>();
        for(Component component: this.children){
            result.add(component.operation());
        }
        return "Branch(" + String.join("+", result) + ')';
    }

    public boolean isComposite(){
        return true;
    }
}
